ports/vcpkg-get-python-packages/x_vcpkg_get_python_packages.cmake

```
@@ -0,0 +1,82 @@
+#[===[.md:
+# x_vcpkg_get_python_packages
+
+Experimental
+Retrieve needed python packages
+
+## Usage
+```cmake
+x_vcpkg_get_python_packages(
+    <PYTHON_DIR>
+    PYTHON_EXECUTABLE <path to python binary>
+    PACKAGES <packages to aqcuire>...
+)
+```
+## Parameters
+
+### PYTHON_EXECUTABLE
+Full path to the python executable 
+
+### PACKAGES
+List of python packages to acquire
+
+#]===]
+if(Z_VCPKG_GET_PYTHON_PACKAGES_GUARD)
+    return()
+endif()
+set(Z_VCPKG_GET_PYTHON_PACKAGES_GUARD ON CACHE INTERNAL "guard variable")
+
+function(x_vcpkg_get_python_packages PYTHON_DIR )
+    cmake_parse_arguments(PARSE_ARGV 0 arg "" "PYTHON_EXECUTABLE" "PACKAGES")
+    
+    if(NOT arg_PYTHON_EXECUTABLE)
+        message(FATAL_ERROR "${CMAKE_CURRENT_FUNCTION} requires parameter PYTHON_EXECUTABLE!")
+    endif()
+    if(NOT arg_PACKAGES)
+        message(FATAL_ERROR "${CMAKE_CURRENT_FUNCTION} requires parameter PACKAGES!")
+    endif()
+    if(NOT arg_PYTHON_DIR)
+        get_filename_component(arg_PYTHON_DIR "${arg_PYTHON_EXECUTABLE}" DIRECTORY)
+    endif()
+
+    if (WIN32)
+        set(PYTHON_OPTION "")
+    else()
+        set(PYTHON_OPTION "--user")

```

> --user  Install to the Python user install directory for your platform. 
        Typically ~/.local/, or %APPDATA%\Python on Windows. 
        (See the Python documentation for site.USER_BASE for full details.)
        
This still modifies the user's system. I think we should be using a virtual environment instead.

---

This script is not installing anything if python3 outside of vcpkg is used. 
venv could only help installing packages if python3 outside of vcpkg is used. So we wouldn't throw an error.

---

also, this should be:
```suggestion
        vcpkg_list(SET python_user_param "--user")
```